### PR TITLE
Added new enums from BACnet addenda

### PR DESCRIPTION
**Pull request by skarg**
Added new BacnetErrorCodes enums
Added new BacnetNetworkMessageTypes enums
Added new BacnetObjectTypes enums
Added new BacnetPropertyIds enums
Added new BacnetReliability enums
Added new BacnetServicesSupported enums
Added new BacnetUnconfirmedServices enums
Added new BacnetUnitsId enums
Added new BacnetEventTypes enums

(copied from BACnet Stack in C hosted at SF.net and GitHub)

Updated README to reflect where most of the enums originated.

### CODE REVIEW

**Review by pgolinski — CHANGES_REQUESTED** [review]
Build of the project failes because these enums are duplicated:

UNITS_STANDARD_CUBIC_FEET_PER_DAY
UNITS_MILLION_STANDARD_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_STANDARD_CUBIC_FEET_PER_DAY
UNITS_STANDARD_CUBIC_FEET_PER_DAY
UNITS_MILLION_STANDARD_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_STANDARD_CUBIC_FEET_PER_DAY
UNITS_STANDARD_CUBIC_FEET_PER_DAY
UNITS_MILLION_STANDARD_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_STANDARD_CUBIC_FEET_PER_DAY
UNITS_STANDARD_CUBIC_FEET_PER_DAY
UNITS_MILLION_STANDARD_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_CUBIC_FEET_PER_DAY
UNITS_THOUSAND_STANDARD_CUBIC_FEET_PER_DAY

File: Base/BacnetErrorCodes.cs [hunk 1]
```diff
@@ -164,9 +164,77 @@ public enum BacnetErrorCodes
...
+    ERROR_CODE_TCP_ERROR = 197,
+    ERROR_CODE_IP_ADDRESS_NOT_REACHABLE = 198,
+    ERROR_CODE_IP_ERROR = 199,
+    ERROR_CODE_CERTIFICATE_EXPIRED = 200,
+    ERROR_CODE_CERTIFICATE_INVALID = 201,
+    ERROR_CODE_CERTIFICATE_MALFORMED = 202,
+    ERROR_CODE_CERTIFICATE_REVOKED = 203,
+    ERROR_CODE_UNKNOWN_SECURITY_KEY = 204,
+    ERROR_CODE_REFERENCED_PORT_IN_ERROR = 205,
+    MAX_BACNET_ERROR_CODE = 205,
```
Comment (gralin) [inline Base/BacnetErrorCodes.cs:236]: should be `MAX_BACNET_ERROR_CODE = 206` if following the convention

File: Base/BacnetObjectTypes.cs [hunk 2]
```diff
@@ -58,6 +58,15 @@ public enum BacnetObjectTypes : uint
...
     OBJECT_BINARY_LIGHTING_OUTPUT = 55, /* Addendum 135-2012az */
+    OBJECT_NETWORK_PORT = 56,           /* Addendum 135-2012az */
+    OBJECT_ELEVATOR_GROUP = 57,         /* Addendum 135-2012aq */
+    OBJECT_ESCALATOR = 58,              /* Addendum 135-2012aq */
+    OBJECT_LIFT = 59,                   /* Addendum 135-2012aq */
+    OBJECT_STAGING = 60,                /* Addendum 135-2016bd */
+    OBJECT_AUDIT_LOG = 61,              /* Addendum 135-2016bi */
+    OBJECT_AUDIT_REPORTER = 62,         /* Addendum 135-2016bi */
+    OBJECT_COLOR = 63,                  /* Addendum 135-2020ca */
+    OBJECT_COLOR_TEMPERATURE = 64,      /* Addendum 135-2020ca */
```
Comment (gralin) [inline Base/BacnetObjectTypes.cs:69]: Also update `MAX_ASHRAE_OBJECT_TYPE = 65`

File: Base/BacnetUnconfirmedServices.cs [hunk 3]
```diff
@@ -14,10 +14,15 @@ public enum BacnetUnconfirmedServices : byte
     SERVICE_UNCONFIRMED_UTC_TIME_SYNCHRONIZATION = 9,
     /* addendum 2010-aa */
     SERVICE_UNCONFIRMED_WRITE_GROUP = 10,
+    /* addendum 2012-aq */
+    SERVICE_UNCONFIRMED_COV_NOTIFICATION_MULTIPLE = 11,
+    /* addendum 2016-bz */
+    SERVICE_UNCONFIRMED_WHO_AM_I = 13,
+    SERVICE_UNCONFIRMED_YOU_ARE = 14,
```
Comment (gralin) [inline Base/BacnetUnconfirmedServices.cs:21]: Please use consecutive numbering here (missing `12`)

File: Base/BacnetServicesSupported.cs [hunk 4]
```diff
@@ -41,6 +43,7 @@ public enum BacnetServicesSupported
     SERVICE_SUPPORTED_REQUEST_KEY = 25,
     SERVICE_SUPPORTED_I_AM = 26,
     SERVICE_SUPPORTED_I_HAVE = 27,
+    /* Unconfirmed Services */
     SERVICE_SUPPORTED_UNCONFIRMED_COV_NOTIFICATION = 28,
```
Comment (gralin) [inline Base/BacnetServicesSupported.cs:47]: Unconfirmed Services start from 26, can you push the comment up?

File: Base/BacnetServicesSupported.cs [hunk 5]
```diff
@@ -49,8 +52,9 @@ public enum BacnetServicesSupported
     SERVICE_SUPPORTED_UTC_TIME_SYNCHRONIZATION = 36,
     SERVICE_SUPPORTED_WHO_HAS = 33,
     SERVICE_SUPPORTED_WHO_IS = 34,
+    SERVICE_SUPPORTED_UNCONFIRMED_AUDIT_NOTIFICATION = 46,
```
Comment (gralin) [inline Base/BacnetServicesSupported.cs:55]: We are also missing other services here that you added in `Base/BacnetUnconfirmedServices.cs` like who-am-i and you-are, can you include them here too? And the other way around, this audit-notification service is not included in that file but is present here.

I don't have access to those addendums beyond 2016 so I trust you here.

File: Base/Enums/BacnetEventTypes.cs [hunk 6]
```diff
@@ -14,6 +14,16 @@ public enum BacnetEventTypes
     EVENT_EXTENDED = 9,
     EVENT_BUFFER_READY = 10,
     EVENT_UNSIGNED_RANGE = 11,
+    EVENT_ACCESS_EVENT = 13,
```
Comment (gralin) [inline Base/Enums/BacnetEventTypes.cs:17]: Can you add comment, that 12 is reserved for future addenda?

**Review by gralin — CHANGES_REQUESTED** [review]

File: Base/Enums/BacnetEventTypes.cs (hunk 6 above)
Comment (skarg) [reply Base/Enums/BacnetEventTypes.cs:18]: sure!

File: Base/BacnetServicesSupported.cs (hunk 5 above)
Comment (skarg) [reply Base/BacnetServicesSupported.cs:57]: BACnet addenda are free to download from ashrae.org and bacnet.org websites.
UnconfirmedAuditNotification is here: http://www.bacnet.org/Addenda/Add-135-2016bi.pdf
Thanks for noticing the missing Who-Am-I and You-Are services here!

Comment (gralin) [reply Base/BacnetServicesSupported.cs:57]: Thanks, didn't know that! I thought we have to buy the standard again to get everything updated (the one I have access to is from 2016).

**Review by gralin — APPROVED** [review]

**Pull request merged**